main/adc/adc.h

```diff
@@ -0,0 +1,18 @@
+#ifndef ADC_H_
+#define ADC_H_
+
+#include <esp_err.h>
+
+#define ADC1_PORTS 1
+
+struct ADC_data
```

Comment:
Please use camel_case, so rename it to `adc_data`
Possibly make this into a typedef like so;

`typedef struct { ... } adc_data_t`